src/contour/display/TerminalWidget.cpp

```diff
@@ -1260,12 +1260,9 @@ void TerminalWidget::scheduleRedraw()
         _lastHistoryLineCount = currentHistoryLineCount;
     }
 
-    if (setScreenDirty())
-    {
-        // QCoreApplication::postEvent(this, new QEvent(QEvent::UpdateRequest));
-        if (window())
-            post([this]() { window()->update(); });
-    }
+    // QCoreApplication::postEvent(this, new QEvent(QEvent::UpdateRequest));
```

Comment:
So your change here is simply to not mark the screen as dirty, right?

Reply:
which simply calls `RenderStateManager.touch()`, so that itself doesn't have any side effects.
Since this however is done using atomic bool operations, I really wonder why that has an effect. I sadly cannot reproduce this (maybe my PC too fast? who knows... ah, and it's ARM64 on my end, who knows)....

I looked into into where we use that information. Looks like an optimization relic where I tried to do as minimal renders as possible. Looking at `paint()` impl, we are always rendering, regardless of the current dirty-state on the GUI side.

I am fine in dropping this line (or have it commented out?), but in the long run, we should evaluate if we do  still need this design idea of minimizing render operations (does it still have an effect).

Reply:
i will try to get rid of this atomic